misc/python/materialize/scalability/schema.py

```diff
@@ -0,0 +1,67 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from enum import Enum, auto
+from typing import Optional
+
+
+class Source(Enum):
+    TABLE = auto()
+
+
+class TransactionIsolation(Enum):
+    SERIALIZABLE = "serializable"
+    STRICT_SERIALIZABLE = "strict serializable"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class Schema:
+    def __init__(
+        self,
+        source: Source = Source.TABLE,
+        schema: str = "scalability",
+        index: bool = True,
+        transaction_isolation: Optional[TransactionIsolation] = None,
+        cluster: Optional[str] = None,
+    ) -> None:
+        self.schema = schema
+        self.source = source
+        self.index = index
+        self.transaction_isolation = transaction_isolation
+        self.cluster = cluster
+
+    def init_sqls(self) -> list[str]:
+        init_sqls = self.connect_sqls() + [
+            f"DROP SCHEMA IF EXISTS {self.schema} CASCADE;",
+            f"CREATE SCHEMA {self.schema};",
+            "DROP TABLE IF EXISTS t1;",
+        ]
+        if self.source == Source.TABLE:
+            init_sqls.append("CREATE TABLE t1 (f1 INTEGER);")
+        # else:
+        #    return NotImplementedError
```

Comment:
Commented-out code left over

Reply:
He he I was trying to be clever here but the type checker says that for an enum with 1 possible value, the `else` is unreachable. Will remove.

Reply:
Oh, ok. That's not nice from the type checker. Will it also complain if a switch-case is used? (If I am not mistaken, I used such stuff in the output-consistency framework to get aware when a new enum value reaches a statement not prepared for that.)